enarx-keep-sgx-shim/src/handler.rs

```diff
@@ -644,4 +644,28 @@ impl<'a> Handler<'a> {
 
         trusted.len() as u64
     }
+
+    // Do clock_gettime syscall
+    pub fn clock_gettime(&mut self) -> u64 {
+        self.trace("clock_gettime", 2);
+
+        let clk_id = self.aex.gpr.rdi.raw();
+        let ts = self.aex.gpr.rsi.raw() as *mut libc::timespec;
+
+        unsafe {
+            let (_, untrusted, _) = self.block.buf.align_to::<libc::timespec>();
+            let uaddr = untrusted.as_ptr();
+            let result = self.proxy(request!(libc::SYS_clock_gettime => clk_id, uaddr));
+            match result {
+                Ok([ret, _]) => {
+                    let ret = ret.raw() as u64;
```

Comment:
If I may offer some word of advice here, it is almost always better to use safe wrappers around explicit casts such as `From/Into` or `TryFrom/TryInto` rather than the `as` keyword. In this particular case, as far as I can see, `ret.raw() as u64` is fine here, however rewriting is as `let ret: u64 = ret.raw().into()` will force the compiler to trigger a compile error if it is impossible to do a safe cast (i.e., if we were to cast an integer of higher resolution into a smaller one such as `u64` into `u32`).

Reply:
Thanks! Got it fixed in the latest change.